### PR TITLE
fix: DateSelector selector validation message issues

### PR DESCRIPTION
**Pull request by YuvalMasada**
@NoamGaash - This time at the right direction
I've added 2 short fixes.

1. For wrong validation message I added a different message for the specific case. The message will pop for min/max, but I think that the max message will never appear( so maybe it would be better to delete it? not 100% sure ).
2. To make the message re-render when language changed I've extend the useMemo dependency array with t function that should be reassign as the DateSelector component been re-rendered.

### CODE REVIEW

File: src/pages/components/DateSelector.tsx [hunk 1]
```diff
@@ -20,7 +21,7 @@ export function DateSelector({ time, onChange, customLabel, minDate }: DataAndTi
         return ''
       }
     }
-  }, [error])
+  }, [error, t])
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:24]: Maybe it would be better if we'll just remove the `useMemo` hook

**Review by NoamGaash — APPROVED** [review]

File: src/pages/components/DateSelector.tsx (hunk 1 above)
Comment (YuvalMasada) [reply src/pages/components/DateSelector.tsx:24]: @NoamGaash 
It's a good idea to remove it.
while removing it I had thought about 2 how to implement it. 

1.  Just removing the useMemo and use the function.
2.  Take the function out of the component( and remove the useTranslation also) - will look better. 
does it matter which approach I go forward with? I did both of them and it works.

Comment (NoamGaash) [reply src/pages/components/DateSelector.tsx:24]: That's up to you
I'd like to make a third suggestion - 
1. the function could return the translation key
2. the function can be inside the scope of the component, or underneath

File: src/pages/components/DateSelector.tsx [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
 import { DataAndTimeSelectorProps } from './utils/dateAndTime'
 
+type errorMessageTypes = '' | 'bug_date_alert' | 'bug_date_invalid_format'
+const getErrorMessage = (error: DateValidationError): errorMessageTypes => {
+  let message: errorMessageTypes = ''
+  switch (error) {
+    case 'maxDate':
+    case 'minDate':
+      message = t('bug_date_alert')
+      break
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:14]:
```suggestion
      return 'bug_date_alert'
```

File: src/pages/components/DateSelector.tsx [hunk 3]
```diff
@@ -1,27 +1,33 @@
...
+    case 'invalidDate': {
+      message = t('bug_date_invalid_format')
+      break
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:17]:
```suggestion
      return 'bug_date_invalid_format'
```

File: src/pages/components/DateSelector.tsx [hunk 4]
```diff
@@ -1,27 +1,33 @@
...
+    }
+
+    default: {
+      message = ''
+    }
+  }
+  return message
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:24]:
```suggestion
    }
```

File: src/pages/components/DateSelector.tsx [hunk 5]
```diff
@@ -1,27 +1,33 @@
...
+import { useState } from 'react'
 import { DatePicker } from '@mui/x-date-pickers/DatePicker'
-import { useTranslation } from 'react-i18next'
+import { t } from 'i18next'
 import { DateValidationError } from '@mui/x-date-pickers'
 import { DataAndTimeSelectorProps } from './utils/dateAndTime'
 
+type errorMessageTypes = '' | 'bug_date_alert' | 'bug_date_invalid_format'
+const getErrorMessage = (error: DateValidationError): errorMessageTypes => {
+  let message: errorMessageTypes = ''
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:9]:
```suggestion
```

File: src/pages/components/DateSelector.tsx [hunk 6]
```diff
@@ -1,27 +1,33 @@
-import { useState, useMemo } from 'react'
+import { useState } from 'react'
 import { DatePicker } from '@mui/x-date-pickers/DatePicker'
-import { useTranslation } from 'react-i18next'
+import { t } from 'i18next'
 import { DateValidationError } from '@mui/x-date-pickers'
 import { DataAndTimeSelectorProps } from './utils/dateAndTime'
 
+type errorMessageTypes = '' | 'bug_date_alert' | 'bug_date_invalid_format'
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:7]:
```suggestion
type errorMessageTypes = undefined | 'bug_date_alert' | 'bug_date_invalid_format'
```

File: src/pages/components/DateSelector.tsx [hunk 7]
```diff
@@ -1,27 +1,33 @@
...
-      case 'invalidDate': {
-        return t('bug_date_invalid_format')
-      }
 
-      default: {
-        return ''
-      }
-    }
-  }, [error, t])
+  const errorMessage = getErrorMessage(error)
```
Comment (NoamGaash) [inline src/pages/components/DateSelector.tsx:30]:
```suggestion
  const errorMessage = t(getErrorMessage(error))
```

**Review by NoamGaash — APPROVED** [review]
I made a suggestion - it will let you keep using `useTranslation` which is better approach as it will trigger re-render on language change

**Comment by YuvalMasada** [comment]
@NoamGaash 
At first I did something like your suggestion, but then typescript got mad at me. 
It doesn't like that getErrorMessage can return undefined( the **t** function we got from the useTranslation hook). 
I'll be happy to understand why typescript complain about it.

**Comment by NoamGaash** [comment]
@YuvalMasada I'll try

**Pull request merged**